File: leetcode/1266. Minimum Time Visiting All Points/Solution.java
class Solution {
    /**
     * The minimum time to move between two points (x1, y1) and (x2, y2)
     * is the maximum of the absolute differences of their coordinates.
     */
    public int minTimeToVisitAllPoints(int[][] points) {
        int totalTime = 0;
        for (int i = 0; i < points.length - 1; i++) {
            int x1 = points[i][0];
            int y1 = points[i][1];
            int x2 = points[i + 1][0];
            int y2 = points[i + 1][1];
            totalTime += Math.max(Math.abs(x2 - x1), Math.abs(y2 - y1));
        }
        return totalTime;
    }

    public static void main(String[] args) {
        Solution sol = new Solution();

        // Example 1
        int[][] points1 = { { 1, 1 }, { 3, 4 }, { -1, 0 } };
        System.out.println("Example 1: " + sol.minTimeToVisitAllPoints(points1)); // Expected: 7

        // Example 2
        int[][] points2 = { { 3, 2 }, { -2, 2 } };
        System.out.println("Example 2: " + sol.minTimeToVisitAllPoints(points2)); // Expected: 5
    }
}

File: leetcode/1266. Minimum Time Visiting All Points/solution.py
from typing import List

class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        """
        The minimum time to move between two points (x1, y1) and (x2, y2)
        is the maximum of the absolute differences of their coordinates:
        max(abs(x2 - x1), abs(y2 - y1)).
        This is because we can move diagonally to cover both dimensions 
        simultaneously as much as possible, then move horizontally or vertically.
        """
        total_time = 0
        for i in range(len(points) - 1):
            x1, y1 = points[i]
            x2, y2 = points[i+1]
            total_time += max(abs(x2 - x1), abs(y2 - y1))
        return total_time

if __name__ == "__main__":
    sol = Solution()
    # Example 1
    print(f"Example 1: {sol.minTimeToVisitAllPoints([[1,1],[3,4],[-1,0]])}") # Expected: 7
    # Example 2
    print(f"Example 2: {sol.minTimeToVisitAllPoints([[3,2],[-2,2]])}")       # Expected: 5
